src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,219 @@
+ */
+export class MissingTerms extends Component {
+  static ID = 'MissingTerms';
+  static options: IMissingTermsOptions = {
+    /**
+     * Whether to allow the end-user to click a missing term to re-inject it in the query as
+     * an exact phrase match (i.e., as an expression between double quote characters).
+     *
+     * **Default:** `true`
+     */
+    clickable: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+    /**
+     * The text to display before missing terms.
+     *
+     * **Default:** The localized string for `Missing`.
+     */
+    caption: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'Missing' }),
+    /**
+     * The maximum number of missing term to be displayed
+     *
+     * **Default:** `5`
+     * **Minimum value:** `1`
+     */
+    numberOfTerms: ComponentOptions.buildNumberOption({
+      defaultValue: 5,
+      min: 1
+    })
+  };
+
+  static doExport = () => {
+    exportGlobally({
+      MissingTerms: MissingTerms
+    });
+  };
+
+  // Used to split terms and phrases. Match character that can separate words or caracter for Chinese, Japanese and Korean.
+  // Han: Unicode script for Chinesse character
+  // We only need to import 1 Asian, charcaters script because what is important here is the space between the caracter and any script will contain it
+  private wordBoundary = '(([\\p{Han}])?([^(\\p{Latin}-)])|^|$)';
+  private termForcedToAppear: string[];
+
+  /**
+   * Creates a new `MissingTerms` component instance.
+   * @param element The element on which to instantiate the component.
+   * @param options The configuration options for the component.
+   * @param bindings The bindings required by the component to function normally. If not set, these will be automatically resolved (with a slower execution time).
+   * @param result The query result item to associate the component with.
+   */
+  constructor(
+    public element: HTMLElement,
+    public options?: IMissingTermsOptions,
+    bindings?: IComponentBindings,
+    public result?: IQueryResult
+  ) {
+    super(element, MissingTerms.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerms, options);
+    this.addMissingTerms();
+  }
+  /**
+   *Returns all original basic query expression terms that were not matched by the result item the component instance is associated with.
+   */
+  public get missingTerms(): string[] {
+    return this.result.absentTerms.filter(term => {
+      const regex = this.createWordBoundaryDelimitedRegex(term);
+      return regex.test(this.queryStateModel.get('q'));
+    });
+  }
+
+  /**
+   * Inject a term in the advanced query.
+   * This will make sure the term that are in the advance query appears in the results
+   */
+  public addTermForcedToAppear(term: string) {
+    if (this.missingTerms.indexOf(term) === -1) {
+      this.logger.warn(
+        'The term to re-inject is not present in the missing terms',
```

Comment:
I'd rephrase the message as follows:

> Method execution aborted because the term to inject in `aq` ("${term}") is not a missing term.
> Allowed missing terms: ${this.missingTerms.toString()}.